src/Async.js

```diff
@@ -208,8 +216,19 @@ export default class Async extends Component {
 		this.loadOptions(inputValue, this.state.page + 1);
 	}
 
+	onChange(value) {
+		this.props.onChange(value);
+
+		if (this.props.pagination) {
+			let remainingOptions = this.state.options.length - value.length;
```

Comment:
What is `value` here? It can be a string or an object as well as an array. When it is not an array, this doesn't make much sense.

Reply:
ah, you're correct. these fixes were specifically for my use case which was multi-select async pagination. let me test out some of the other edge cases and get back to you with another commit.